File: tests/XMP/global-view/shadow-reflect/F/full2.f90
program test

!$xmp nodes p(2,2)

!$xmp template t(4,4)
!$xmp distribute t(block,block) onto p

  integer a(4,4)
!$xmp align a(i,j) with t(i,j)
!$xmp shadow a(*,*)

  integer :: result = 0

!$xmp loop (i,j) on t(i,j)
  do j = 1, 4
     do i = 1, 4
        a(i,j) = i * 10 + j
     end do
  end do

!$xmp reflect (a)

  do j = 1, 4
     do i = 1, 4
        if (a(i,j) /= i * 10 + j) then
           result = 1
        end if
     end do
  end do

!$xmp reduction (+:result)

!$xmp task on p(1,1)
  if (result == 0) then
     write(*,*) "PASS"
  else
     write(*,*) "ERROR"
     call exit(1)
  endif
!$xmp end task

end program test

File: tests/XMP/global-view/shadow-reflect/C/full2.c
#include <stdio.h>
#include <stdlib.h>

#pragma xmp nodes p[2][2]

#pragma xmp template t[4][4]
#pragma xmp distribute t[cyclic][cyclic] onto p

int a[4][4];
#pragma xmp align a[i][j] with t[i][j]
#pragma xmp shadow a[*][*]

int result = 0;

int main(){

#pragma xmp loop on t[i][j]
  for (int i = 0; i < 4; i++){
    for (int j = 0; j < 4; j++){
      a[i][j] = i * 10 + j;
    }
  }

#pragma xmp reflect (a)

  for (int i = 0; i < 4; i++){
    for (int j = 0; j < 4; j++){
      if (a[i][j] != i * 10 + j) result = 1;
    }
  }

#pragma xmp reduction(+:result)

#pragma xmp task on p[0][0]
  {
    if (result == 0){
      printf("PASS\n");
    }
    else{
      fprintf(stderr, "ERROR\n");
      exit(1);
    }
  }

  return 0;

}
